src/experimental/java/org/quiltmc/loader/api/ConvertibleModMetadata.java

```diff
@@ -0,0 +1,11 @@
+package org.quiltmc.loader.api;
+
+/** @deprecated Since this is very much not a good API, and needs changing.
+ *             <p>
+ *             Or at least re-evaluted later on to see if this should be kept as-is. */
+@Deprecated
+public interface ConvertibleModMetadata {
+    net.fabricmc.loader.api.metadata.ModMetadata asFabricModMetadata();
```

Comment:
this *is* gross, but it's fine because abstracting all this away will be its own project

Reply:
This shouldn't be api imo.

Reply:
Actually it would probably be cleaner if this was moved outside of api too :p